src/create_maps.cpp

```diff
@@ -0,0 +1,366 @@
+	sphereColor.insert(pair<vector<double>, double >(it->first,double(d)));
+        poseReach.push_back(it->second);
+	}
+
+    ROS_INFO("No of spheres reachable: %lu",sphereColor.size());
+ 
+
+
+
+
+//Creating maps now
+
+    string path(ros::package::getPath("reuleaux")+"/maps/");
+    if (stat(path.c_str(),&st)!=0)
+	ROS_INFO("Path does not exist. Creating folder for maps");
+	const int dir_err = mkdir(path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+        if (1 == dir_err)
+	{
+    		ROS_INFO("Error creating directory");
+    		exit(1);
+	}
+    
+//Time is taken for saving the filename of the map
+//TODO the filename will be an argument
+//TODO need to include the sphere discretization and pose rotation parameters in the name
+//If the user does not provide the desired name, this will be default name of the map created in the "maps" folder. robot_time_date_resoltion_spdis_porot.h5
+
+    time_t currentTime;
+    struct tm *localTime;
+    time( &currentTime );        
+    localTime = localtime( &currentTime ); 
+   
+    int Day    = localTime->tm_mday;
+    int Month  = localTime->tm_mon + 1;
+    int Year   = localTime->tm_year + 1900;
+    int Hour   = localTime->tm_hour;
+    int Min    = localTime->tm_min;
+    int Sec    = localTime->tm_sec;
+   
+//Creating all the file and group ids and the default file name 
+ 
+    string filename;
+    filename=string(k.getRobotName())+"_"+boost::lexical_cast<std::string>(Hour)+":"+boost::lexical_cast<std::string>(Min)+"_"+boost::lexical_cast<std::string>(Month)+":"+boost::lexical_cast<std::string>(Day)+":"+boost::lexical_cast<std::string>(Year)+"_"+"r"+str( boost::format("%d") % resolution)+"_"+"sd"+"_"+"rot"+"."+"h5";
+    
+    
+    const char * filepath = path.c_str();
+    const char * name = filename.c_str();
+    char fullpath[100];
+    strcpy(fullpath,filepath);
+    strcat(fullpath,name);
+    ROS_INFO("Saving map %s",filename.c_str());
+    hid_t file, group_reachability, group_capability, group_poses, group_spheres;
+    file = H5Fcreate(fullpath, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+    group_reachability = H5Gcreate(file,"/Reachability",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    group_capability = H5Gcreate(file,"/Capability",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    group_poses = H5Gcreate(group_reachability,"/Reachability/Poses",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    group_spheres = H5Gcreate(group_reachability,"/Reachability/Spheres",H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+
+ 
+//Creating Poses dataset: This process is little complicated. As the resolution decreses, e.g 0.03, the number of poses rises upto 7 digits, so the matrix of double value is not going to support this. We are creating a chunked infinite dataset which will select memory hyperslab and write data to it from buffer. We are using 2 hyperslabs.
+
+    ROS_INFO("Saving poses in reachability map");
+    
+    
+    const hsize_t ndims = 2;
+    const hsize_t ncols = 7;
```

Comment:
the data is 2 dimensional, and 7 is the poses (position ->x,y,z: orientation->x,y,z,w). In the next version, changing it to higher dimension where the sphere points are also stored in the same database for the ease of visualization